urlc98c3b97d2_Trimax_venta_tgz-pJ8-com_gesoftware_venta_structures_map_MapTestJ8/src/com/gesoftware/venta/utility/SerializationUtility.java

```diff
@@ -0,0 +1,93 @@
+package com.gesoftware.venta.utility;
+
+import com.gesoftware.venta.logging.LoggingUtility;
+
+import java.io.*;
+
+/**
+ * SerializationUtility class definition
+ */
+public final class SerializationUtility {
+
+    /* *
+     * METHOD: Transforms object to a byte array if possible
+     * RETURN: Object's byte array representation
+     *  PARAM: [IN] object - object for serialization
+     * AUTHOR: Eliseev Dmitry
+     * */
+    public static byte[] pack(final Object object) {
+        ByteArrayOutputStream bytesArrayOutputStream = new ByteArrayOutputStream();
+        ObjectOutput objectOutput = null;
+
+        /* Serialized object */
+        byte[] serializedObject = null;
+
+        try {
+            objectOutput = new ObjectOutputStream(bytesArrayOutputStream);
+            objectOutput.writeObject(object);
+
+            /* Object serialization */
+            serializedObject = bytesArrayOutputStream.toByteArray();
+        } catch (final IOException e) {
+            LoggingUtility.exception(e);
+        } finally {
+            /* Close object output stream */
+            try {
+                if (objectOutput != null)
+                    objectOutput.close();
+            } catch (final IOException e) {
+                LoggingUtility.error("Error during closing object output stream: " + e.getMessage());
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 35:40, 83:88.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.